### PR TITLE
unicode bug

### PR DESCRIPTION
**Pull request by noisyboiler**
this is far from perfect. just fixes python 2.7 and is a way of appreciating the problem.

- test case that exposes a unicode bug
- suggested Python 2.7 fix

Likely still broken on Python 3

### CODE REVIEW

File: wampy/transports/websocket/frames.py [hunk 1]
```diff
@@ -85,7 +85,14 @@ def __init__(self, bytes):
         self.body = bytes
 
     def __len__(self):
-        return len(self.body)
```
Comment (noisyboiler) [inline wampy/transports/websocket/frames.py]: @titilambert 
this was part of the problem

File: wampy/transports/websocket/frames.py [hunk 2]
```diff
@@ -127,8 +132,10 @@ def generate_mask(self, mask_key, data):
         if data is None:
             data = ""
 
+        data_bytes = bytearray(data, 'utf-8')
+
         _m = array.array("B", mask_key)
-        _d = array.array("B", data.encode())
```
Comment (noisyboiler) [inline wampy/transports/websocket/frames.py]: @titilambert 
part of the problem.

File: wampy/transports/websocket/frames.py [hunk 3]
```diff
@@ -236,7 +246,7 @@ def __init__(self, bytes):
 
         self.opcode = bytes[0] & 0b1111
         try:
-            self.payload = json.loads(str(self.body.decode()))
```
Comment (noisyboiler) [inline wampy/transports/websocket/frames.py]: @titilambert 
again, part of the problem

File: wampy/transports/websocket/frames.py [hunk 4]
```diff
@@ -85,7 +85,14 @@ def __init__(self, bytes):
         self.body = bytes
 
     def __len__(self):
-        return len(self.body)
+        # UTF-8 is an unicode encoding which uses more than one byte for
+        # special characters. calculating the length needs consideration.
+        try:
+            unicode_body = self.body.decode("utf-8")
```
Comment (noisyboiler) [inline wampy/transports/websocket/frames.py]: i think this is okay. WAMP protocol dictates this encoding. possibly though i could enforce this at an earlier point in the serialisation process.

**Comment by titilambert** [comment]
@noisyboiler Did you look at the code in autobahn ? Maybe you can kind of "copy/paste" the frame/payload part ?

**Comment by titilambert** [comment]
@noisyboiler 
See here: https://github.com/crossbario/autobahn-python/blob/master/autobahn/websocket/protocol.py#L2357
They process binaries and texts with different ways... (utf-8 text is binary) this could be an answer

**Comment by noisyboiler** [comment]
thanks @titilambert. i'll defo take a read. autobahn can be a tricky project to get your head around though, hence writing wampy in the first place :)

**Comment by noisyboiler** [comment]
something wrong with the Travis build of python 3.4. doesn't look to be the fault of wampy.

**Comment by titilambert** [comment]
@noisyboiler I just test my own project with wampy revision: c3f19cb
And this is working ! (with python 3.6.3)

**Comment by noisyboiler** [comment]
that's great news! i need to update my test assertion for python 2 compat. then I'll do a release.

**Comment by titilambert** [comment]
@noisyboiler nice ! thanks ! I can't wait !

**Pull request merged**